ramblings/scenario_variables_etc.md

```diff
@@ -0,0 +1,94 @@
+# Scenario / variables / other filters etc
+
+At the moment we have scenarios and years hard-coded as things that can change the output on the map. We want to generalise this.
+
+I'm struggling to think of a decent name for this ... manifesto, but I'm going formalise what we're doing when we're painting pretty colours on the map a little bit. Heckling is requested and desired.
+
+# What is a model / survey data?
+
+I'm going to be lazy and pretend that predictions and observations are the same (which philosophically makes sense to me - observations are a guess at ground truth). So I'll call them all models.
+
+What is a model? It maps a series of independent variables (things you can change) to dependent variables (things you measure or predict).
+
+```
+model_[dependent_variable](independent_variables...) = {zoneID => some number}
+```
+
+So, for example,
+
+```
+model_kgco2perKM(scenario="the-squirrels-fight-back", year=1862, transportMode="jetpack", passengerAges="18-27") = {... => 9001}
+```
+
+Our key assumption is that each model has consistent units (i.e. different sets of criteria are directly comparable for the same dependent variable). We also assume that addition makes sense for all outputs.
+
+# How does this map to CSVs?
+
+Each CSV column (aside from the origin and destination zone IDs) corresponds to one set of `dependent_variable-independent_variables...`. Using the example from above:
+
+```
+model_kgco2perKM(scenario="the-squirrels-fight-back", year=1862, transportMode="jetpack", passengerAges="18-27") = {... => 9001}
+
+# corresponds to the column name
+kgco2perKM_the-squirrels-fight-back_1862_jetpack_18-27
+```
+
+Open question: how do we deal with missing independent variables? i.e. holes in the domain of the model? E.g. `scenario=millenium_bug` may not have `passengerAges`.
+
+Suggestion: if we want to support this, use a simple `key=value` format for column names, e.g. `kgco2perKM_scenario=the-squirrels-fight-back_transportMode=jetpack_...`.
+
+# How do we pick these in the web app?
+
+"variable" (the dependent one) is the only thing we are certain will exist for every project, so it should probably go at the top as the first filter.
+
+Then each independent variable is listed below as a filter with a drop-down menu for each. If "compare" is ticked, another panel appears with little "chain link" icons to allow us to link/unlink independent variables for criteria.
+
+For simplicity, values will be sorted alphanumerically in the drop-down menu to deal with categorical and continuous variables comfortably.
```

Comment:
In GCVT the client wanted to specify the order of scenarios and variables manually. I think this will still be required, though we could default to lexical/alphanumeric ordering if no other order is specified.

Reply:
We're going to want to provide a way to provide pretty names for all the (both dependent and independent) variables (e.g. kgco2perkm -> "kg CO2 per km"). If we do alphanumeric sorting based on the un-pretty names then we can just tell people to do e.g. `1-the-squirrels-fight-back`; `2-the-squirrels-dont-like-peanut-butter`. I think that's reasonably intuitive provided that it's documented